tests/testthat/testEdgeThresh.R

```diff
@@ -0,0 +1,251 @@
+
+    # Check that the size of the largest connected component has not changed
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+
+    # Check that overall size has not changed
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 10)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 20)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 50)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 100)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+
+    ###
+    myPermutation <- .permuteGraph(myEGG, Q = 200)
+
+    expect_true(igraph::is.simple(myPermutation))
+    expect_equal(max(igraph::components(myPermutation)$csize), nLargest)
+    expect_equal(igraph::vcount(myPermutation), nVertices)
+
+})
+# ((Is it possible to pass permuted graphs from the previous test section
+# to the next one, if they pass? This would minimize a lot of computation))
+test_that("a sane input to edgeThresh gives an expected output", {
+    # Provide small input data, provide small output data.
+    #   (Note that the output needs to be independently verifiably correct.
+    #    Don't run an erroneous function, and then test against the erroneous
+    #    output you received when you ran your function for the first time.)
+    # Try this with important variations of parameters (reasonable, not
+    #    combinatorially exhaustive).
+    # Cover your code.
+
+    # ((Is graph.attributes(gWithThresh)$delta the correct way of retrieving
+    # delta values from graph attributes?)))
+    ### Check edgeThresh with different valid input values
+    ### For each check, length deltaVector should be Lmax - 1, numeric, and
+    ### between 0 and 1.
+    ### Check that the input graph is not changed
+    ### Test with N & Q = 1, 5, 10, Lmax 10 & 20
+    ### ((Test with N = 10, Q = 1 -> is there a limit to how many permutations
+    # there can be if only |E| edge swaps take place?))
+
+    gWithThresh <- edgeThresh(myEGG, Lmax = 20, N = 1, Q = 1)
+    deltaVector <- igraph::graph.attributes(gWithThresh)$delta
+
+    # Check that the input graph is not changed
+    # ((what is the fastest way to check this?))
+    expect_true(isomorphic(gWithThresh, myEGG))
+
```

Comment:
Hang on: are you claiming that your edge permutations produce isomorphic graphs?